### DISCUSSION TITLE
Does React's closure trap exist in Dioxus?

### DISCUSSION THREAD

**Discussion by qouteall** [DioxusLabs/dioxus#1316] [Q&A] [answered]
I am new to Dioxus. There is the closure trap in React hooks https://betterprogramming.pub/understanding-the-closure-trap-of-react-hooks-6c560c408cde

As Dioxus is similar to React, does the similar closure trap exists in Dioxus when using `use_state`? Will `use_ref` in Dioxus solve the closure trap?

**Comment by ealmloff** [comment, marked as answer]
It does exist, see https://github.com/DioxusLabs/dioxus/issues/619. You can get the current value with the `current()` function on use_state or the value of use_ref is always up to date
